src/main/java/cms/gov/madie/measure/resources/AdminController.java

```diff
@@ -123,6 +124,60 @@ public ResponseEntity<MeasureTestCaseValidationReportSummary> validateAllMeasure
             .build());
   }
 
+  // This endpoint is used to reset the validation queue for QI Core v6 measures test cases
+  // that are in validating state
+  @PutMapping("/measures/test-cases/restart-validation")
+  @PreAuthorize("#request.getHeader('api-key') == #apiKey")
+  public void resetTestCaseValidationQueue(
+      HttpServletRequest request,
+      @Value("${admin-api-key}") String apiKey,
+      Principal principal,
+      @RequestHeader("Authorization") String accessToken) {
+
+    log.info(
+        "User [{}] - Starting admin task to place QI Core v6 testcases back on the validation queue",
+        principal.getName());
+    List<Measure> measureList =
+        measureRepository.findAllByModel(ModelType.QI_CORE_6_0_0.getValue());
+
+    if (CollectionUtils.isNotEmpty(measureList)) {
+      measureList.forEach(
+          measure -> {
+            if (CollectionUtils.isNotEmpty(measure.getTestCases())) {
+              measure
+                  .getTestCases()
+                  .forEach(
+                      testCase -> {
+                        if (TestCaseValidationStatus.PENDING
+                            .toString()
+                            .equalsIgnoreCase(testCase.getValidationStatus())) {
+                          // Submit test case already in PENDING status
+                          testCaseValidationService.submitOnSaveValidationTask(
+                              measure.getId(),
+                              testCase,
+                              accessToken,
+                              ModelType.valueOfName(measure.getModel()));
+                        } else if (TestCaseValidationStatus.VALIDATING
+                            .toString()
+                            .equalsIgnoreCase(testCase.getValidationStatus())) {
+                          measureRepository.setValidationStatusToPending(
+                              testCase.getId(), measure.getId());
+                          // submit the test case after updating its status
+                          testCaseValidationService.submitOnSaveValidationTask(
```

Comment:
The status setter methods return the modified Measure when successful and `null` otherwise. It might be helpful to verify the returned value is not null prior to submitting the Test Case for validation.
```suggestion
                          Measure updatedMeasure = measureRepository.setValidationStatusToPending(
                              testCase.getId(), measure.getId());
                          if (updatedMeasure == null) break; // Issue updating status to Pending.
                          // submit the test case after updating its status
                          testCaseValidationService.submitOnSaveValidationTask(
```

Reply:
thank you, I have updated it